Dockerfile

```diff
@@ -41,11 +41,22 @@ ENV SUPVISD=${SUPVISD:-supervisorctl} \
     DB_PASSWORD=${DB_PASSWORD:-none}\
     SETUP=${SETUP:-0}
 
-RUN ls -lahR /repo \
-    && { echo '/repo/main'; echo '/repo/community'; cat /etc/apk/repositories; } >/etc/apk/repositories.new \
-    && mv /etc/apk/repositories.new /etc/apk/repositories \
-    && apk --update upgrade --no-cache --available\
-    && apk --update add --no-cache curl su-exec tzdata postfix mailx bash openssh supervisor openvas openvas-smb openvas-config gvmd gvm-libs greenbone-security-assistant ospd-openvas \
+RUN { \
+    echo '@custcom /repo/community/'; \
+    echo 'https://dl-5.alpinelinux.org/alpine/v3.14/main/' ; \
+    echo 'https://dl-5.alpinelinux.org/alpine/v3.14/community/' ;\
+    echo 'https://dl-4.alpinelinux.org/alpine/v3.14/main/' ; \
+    echo 'https://dl-4.alpinelinux.org/alpine/v3.14/community/' ;\
+    echo 'https://dl-cdn.alpinelinux.org/alpine/v3.14/main/' ; \
+    echo 'https://dl-cdn.alpinelinux.org/alpine/v3.14/community/' ; \
+    } >/etc/apk/repositories \
+    && cat /etc/apk/repositories \
+    && sleep 5 \
+    && apk update --update-cache \
+    && sleep 5 \
+    && apk upgrade --available \
+    && sleep 5 \
+    && apk add --allow-untrusted curl su-exec tzdata postfix mailx bash openssh supervisor openvas@custcom openvas-smb@custcom openvas-config@custcom gvmd@custcom gvm-libs@custcom greenbone-security-assistant@custcom ospd-openvas@custcom \
```

Comment:
This seems to be missing quite a few packages compared to the existing image

Here are the current ubuntu packages:
<details>
bison
build-essential
ca-certificates
cmake
curl
fakeroot
gcc
gcc-mingw-w64
geoip-database
git
gnutls-bin
graphviz
heimdal-dev
ike-scan
libgcrypt20-dev
libglib2.0-dev
libgnutls28-dev
libgpgme11-dev
libgpgme-dev
libhiredis-dev
libical-dev
libksba-dev
libldap2-dev
libmicrohttpd-dev
libnet1-dev
libnet-snmp-perl
libpcap-dev
libpopt-dev
libradcli-dev
libsnmp-dev
libssh-gcrypt-dev
libunistring-dev
libxml2-dev
net-tools
nmap
nsis
openssh-client
openssh-server
perl-base
pkg-config
postfix
postgresql-12
postgresql-server-dev-12
python3-defusedxml
python3-dev
python3-dialog
python3-lxml
python3-paramiko
python3-pip
python3-polib
python3-psutil
python3-setuptools
redis-server
redis-tools
rpm
rsync
smbclient
sshpass
texlive-fonts-recommended
texlive-latex-extra
uuid-dev    
wapiti
wget
whiptail
xml-twig-tools
xsltproc
</details>

Of course, a lot of these are for package builds, so they're not absolutely required, however there are still some which are optional dependencies for GVM, such as Texlive (which is very heavy), xml tools, samba, fakeroot...

Here is the best I could come up with on my own, there are still things missing, such as wapiti and nsis:
```
    echo 'http://dl-3.alpinelinux.org/alpine/edge/main' >> /etc/apk/repositories && \ 
    echo 'http://dl-3.alpinelinux.org/alpine/edge/community' >> /etc/apk/repositories && \
    apk upgrade --update-cache --available && \
    apk add \
    gvmd greenbone-security-assistant openvas gvm-tools\
    openvas-smb ospd-openvas py3-gvm \
    texlive \
    libxslt xmlstarlet zip \
    fakeroot \
    rpm \
    dpkg \
    gnupg \
    wget \
    openssh sshpass \
    socat \
    net-snmp-tools \
    samba-client \
    py3-lxml \
    gnutls-utils \
    perl-xml-twig
```
These are using the edge community packages (which is probably against the philosophy of this project).

This was in order to get an idea of the weight with all additional packages: 410MB without Texlive, 927MB with Texlive, so you may want to split Texlive off to either an optional switch, or a separate image.

Reply:
Here's the breakdown of the packages if required:

```
# Base GVM Packages
gvmd
greenbone-security-assistant
openvas
gvm-tools
# GVM-Provided extensions
openvas-smb
ospd-openvas
py3-gvm
# Report generation
texlive
# HTML Reports
libxslt
xmlstarlet
zip
# Credential packages
fakeroot
# Credential RPM Packages
rpm
# Credential DEB Packages
dpkg
# Credential EXE Packages
# I do not know what is required for nsis generation :/
# Signature verification / S/MIME support
gnupg
# HTTP Alerts
wget
# SCP Alerts
openssh
sshpass
# Send Alerts
socat
# SNMP Alerts
net-snmp-tools
# SMB Alerts
samba-client
# Tipping point Alerts
py3-lxml
# Certificate generation
gnutls-utils
# Lower sync ram usage
perl-xml-twig
```

Reply:
You are right, some packages missing while not need and only used to build, some othere could be added, to use optional features. I will have a look to that.

Reply:
Here a list of all Packages you mention earlier and there status:

# Base GVM Packages
- [x] gvmd
- [x] greenbone-security-assistant
- [x] openvas
- [x] gvm-tools
# GVM-Provided extensions
- [x] openvas-smb
- [x] ospd-openvas
- [x] py3-gvm
# Report generation
- [ ] texlive
# HTML Reports
- [ ] libxslt
- [ ] xmlstarlet
- [ ] zip
# Credential packages
- [ ] fakeroot
  - Not need
# Credential RPM Packages
- [ ] rpm
  - Not need
# Credential DEB Packages
- [ ] dpkg
  - Not need
# Credential EXE Packages
- I do not know what is required for nsis generation :/
- Not need
# Signature verification / S/MIME support
- [x] gnupg
# HTTP Alerts
- [x] wget
- [x] curl
# SCP Alerts
- [x] openssh
- [ ] sshpass
- [ ] scp
# Send Alerts
- [ ] socat
# SNMP Alerts
- [ ] net-snmp-tools
# SMB Alerts
- [ ] samba-client
# Tipping point Alerts
- [ ] py3-lxml
# Certificate generation
- [x] gnutls-utils
# Lower sync ram usage
- [x] perl-xml-twig

Reply:
For reference, Sourced from: https://github.com/greenbone/gvmd/blob/master/INSTALL.md#prerequisites-for-optional-features